### PR TITLE
92/consistency disable

### PR DESCRIPTION
**Pull request by rawstorage**
not ready to release yet, need stable branch of 9.2Q2SR but code is complete with changes, pep8 and pylint clean, CI tests run locally

### CODE REVIEW

File: PyU4V/replication.py [hunk 1]
```diff
@@ -1093,11 +1103,12 @@ def modify_storage_group_srdf(
...
-        'Establish', 'Split', 'Suspend', 'Restore', 'Resume', 'Failover',
-        'Failback', 'Swap', 'SetBias', and 'SetMode'.
-
+        'Establish', 'EnableConsistency', 'DisableConsistency', 'Split',
+        'Suspend', 'Restore', 'Resume', 'Failover', 'Failback', 'Swap',
+        'SetBias', and 'SetMode'.
         :param storage_group_id: storage group id -- str
-        :param action: the rdf action --str
+        :param action: the rdf action, note enable/disable consistency
+                       feature requires Unisphere 9.2.1.6 or higher --str
```
Comment (helenwalsh) [inline PyU4V/replication.py]: nit - space between -- str

File: PyU4V/replication.py [hunk 2]
```diff
@@ -1313,25 +1326,21 @@ def delete_storagegroup_srdf(
...
 
     def delete_storage_group_srdf(
-            self, storage_group_id, srdf_group_number=None):
+            self, storage_group_id, srdf_group_number, filters=None):
         """Delete srdf pairings for a given storage group.
 
         :param storage_group_id: storage group id -- str
         :param srdf_group_number: srdf group number -- int
+        :param filters: optional boolean filters, half_delete,hop2, force,
+                        symforce,star,bypass,keepR1,keepR2,
```
Comment (helenwalsh) [inline PyU4V/replication.py]: nit - space after the commas

File: PyU4V/tests/ci_tests/test_pyu4v_ci_provisioning.py [hunk 3]
```diff
@@ -478,6 +478,7 @@ def test_get_initiator_list(self):
         initiator_list = self.provisioning.get_initiator_list()
         self.assertIsInstance(initiator_list, list)
         for initiator in initiator_list:
+            print(initiator)
```
Comment (helenwalsh) [inline PyU4V/tests/ci_tests/test_pyu4v_ci_provisioning.py]: remove the print

File: PyU4V/tests/ci_tests/test_pyu4v_ci_replication.py [hunk 4]
```diff
@@ -222,6 +244,26 @@ def test_modify_storage_group_snap_link_by_snap_id(self):
             sg_name, target_sg, snap_name, snap_id, unlink=True)
         self.provisioning.delete_storage_group(target_sg)
 
+    def test_modify_storage_group_snap_link_by_snap_id_relink(self):
+        """Test to cover link snapshot by snap id."""
```
Comment (helenwalsh) [inline PyU4V/tests/ci_tests/test_pyu4v_ci_replication.py]: do you need to mention relink here?

File: PyU4V/tests/unit_tests/test_pyu4v_provisioning.py [hunk 5]
```diff
@@ -2162,3 +2162,33 @@ def test_delete_volume_retry(self):
                 resource_level_id=self.data.array,
                 resource_type=constants.VOLUME,
                 resource_type_id=device_id)
+
+    @mock.patch.object(
+        provisioning.ProvisioningFunctions, 'modify_storage_group',
+        return_value=dict())
+    def test_add_existing_volume_to_storage_group_multi_hop(
+            self, mck_sg_modify):
+        """Test add_existing_volume_to_storage_group for SRDF protected SG."""
```
Comment (helenwalsh) [inline PyU4V/tests/unit_tests/test_pyu4v_provisioning.py:2171]: is this the correct comment?

File: PyU4V/tests/unit_tests/test_pyu4v_provisioning.py [hunk 6]
```diff
@@ -2162,3 +2162,33 @@ def test_delete_volume_retry(self):
...
+        mck_sg_modify.assert_called_once_with(
+            self.data.storagegroup_name,
+            self.data.sg_expand_payload_multi_site)
+
+    @mock.patch.object(
+        provisioning.ProvisioningFunctions, 'modify_storage_group',
+        return_value=dict())
+    def test_add_existing_volume_to_storage_group_single_hop(
+            self, mck_sg_modify):
+        """Test add_existing_volume_to_storage_group for SRDF protected SG."""
```
Comment (helenwalsh) [inline PyU4V/tests/unit_tests/test_pyu4v_provisioning.py:2187]: is this the correct comment?

**Review by helenwalsh — CHANGES_REQUESTED** [review]
Only small things.  Happy if tests run clean.

File: PyU4V/tests/ci_tests/test_pyu4v_ci_replication.py (hunk 4 above)
Comment (rawstorage) [reply PyU4V/tests/ci_tests/test_pyu4v_ci_replication.py]: renamed, copy and paste error.  Corrected in latest commit

File: PyU4V/tests/ci_tests/test_pyu4v_ci_provisioning.py (hunk 3 above)
Comment (rawstorage) [reply PyU4V/tests/ci_tests/test_pyu4v_ci_provisioning.py]: done

File: PyU4V/replication.py (hunk 2 above)
Comment (rawstorage) [reply PyU4V/replication.py]: done

File: PyU4V/replication.py (hunk 1 above)
Comment (rawstorage) [reply PyU4V/replication.py]: done

File: PyU4V/tests/unit_tests/test_pyu4v_provisioning.py (hunk 6 above)
Comment (rawstorage) [reply PyU4V/tests/unit_tests/test_pyu4v_provisioning.py:2187]: it's abreviated to keep on one line bt it is correct it's testing adding a volume to storage group that is SRDF protected with one leg..

**Review by helenwalsh — APPROVED** [review]
If unit tests, ci tests, pep8 and pylint are clean, I approve

**Review by wardc6 — APPROVED** [review]

**Comment by rawstorage** [comment]
Merging CI Test ran clean apart from environmental issue on test_modify_rdf_group_change_label

**Pull request merged**